File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/tasks/CropResultTask.java
package expo.modules.imagepicker.tasks;

import android.content.ContentResolver;
import android.net.Uri;

import org.unimodules.core.Promise;

import java.io.File;

import androidx.annotation.NonNull;
import expo.modules.imagepicker.exporters.CropImageExporter;

public class CropResultTask extends ImageResultTask {
  @NonNull
  private final Uri croppedUri;


  public CropResultTask(@NonNull Promise promise,
                        @NonNull Uri uri,
                        @NonNull ContentResolver contentResolver,
                        @NonNull final File cacheDir,
                        boolean exifData,
                        @NonNull CropImageExporter imageExporter) {
    super(promise, uri, contentResolver, cacheDir, exifData, "", imageExporter);
    croppedUri = uri;
  }

  @Override
  protected File getOutputFile() {
    return new File(croppedUri.getPath());
  }
}

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/tasks/CropResultTask.kt
package expo.modules.imagepicker.tasks

import android.content.ContentResolver
import android.net.Uri
import expo.modules.imagepicker.exporters.CropImageExporter
import org.unimodules.core.Promise
import java.io.File

class CropResultTask(promise: Promise,
                     private val croppedUri: Uri,
                     contentResolver: ContentResolver,
                     cacheDir: File,
                     exifData: Boolean,
                     imageExporter: CropImageExporter) : ImageResultTask(promise, croppedUri, contentResolver, cacheDir, exifData, "", imageExporter) {

  override val outputFile: File
    get() = File(croppedUri.path!!)
}
